### PR TITLE
[WIP] Redis store

### PR DESCRIPTION
**Pull request by pluraliseseverythings**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,71 @@
+def kafka_version = 'kafka_2.11-0.10.1.0'
```
Comment (pluraliseseverythings) [inline Jenkinsfile]: Used to run Benchmark tests on Jenkins. This is going away before merge

**Comment by aelred** [comment]
@pluraliseseverythings how possible would it be to break this up into multiple PRs? e.g. one with the timing stuff and another for Redis. The idea of reviewing this is starting to scare me :P

File: grakn-client/src/main/java/ai/grakn/client/TaskClient.java [hunk 2]
```diff
@@ -19,6 +19,9 @@
 
 package ai.grakn.client;
 
+import ai.grakn.engine.TaskId;
```
Comment (pluraliseseverythings) [inline grakn-client/src/main/java/ai/grakn/client/TaskClient.java]: Just changed exception handling here

File: grakn-engine/src/main/java/ai/grakn/engine/controller/CommitLogController.java [hunk 3]
```diff
@@ -82,19 +81,21 @@ private String submitConcepts(Request req, Response res) {
...
         //Instances to count
         TaskState countingTaskState = UpdatingInstanceCountTask.createTask(this.getClass());
         TaskConfiguration countingTaskConfiguration = UpdatingInstanceCountTask.createConfig(keyspace, req.body());
 
-        // Send two tasks to the pipeline
-        manager.addTask(postProcessingTaskState, postProcessingTaskConfiguration);
-        manager.addTask(countingTaskState, countingTaskConfiguration);
-
-
+        // TODO Use an engine wide executor here
```
Comment (pluraliseseverythings) [inline grakn-engine/src/main/java/ai/grakn/engine/controller/CommitLogController.java]: No point waiting for the first to complete, we can add them concurrently

File: grakn-engine/src/main/java/ai/grakn/engine/controller/GraqlController.java [hunk 4]
```diff
@@ -18,6 +18,25 @@
```
Comment (pluraliseseverythings) [inline grakn-engine/src/main/java/ai/grakn/engine/controller/GraqlController.java]: No change in logic, just instrumented with metrics

File: grakn-core/src/main/java/ai/grakn/engine/TaskId.java [hunk 5]
```diff
@@ -19,16 +19,18 @@
...
+import org.apache.tinkerpop.shaded.jackson.annotation.JsonCreator;
+import org.apache.tinkerpop.shaded.jackson.annotation.JsonProperty;
 
 /**
  * An identifier for a task
  *
  * @author Felix Chapman
  */
 public final class TaskId {
-    private final String value;
```
Comment (pluraliseseverythings) [inline grakn-core/src/main/java/ai/grakn/engine/TaskId.java]: Jackson serialisation doesn't work with final (we can convert this to Autovalue+Builder in the future)

**Comment by pluraliseseverythings** [comment]
!rtg

**Comment by pluraliseseverythings** [comment]
!rtg

**Pull request merged**